wallet/src/main/scala/org/bitcoins/wallet/DLCWallet.scala

```diff
@@ -376,11 +376,70 @@ abstract class DLCWallet extends LockedWallet with UnlockedWalletApi {
     }
   }
 
+  def verifyCETSigs(sign: DLCSign): Future[Boolean] = {
+    for {
+      (dlcDb, dlcOffer, dlcAccept, fundingInputs, outcomeSigs) <- getAllDLCData(
+        sign.eventId)
+
+      client <- clientFromDb(dlcDb,
+                             dlcOffer,
+                             dlcAccept,
+                             fundingInputs,
+                             outcomeSigs)
+    } yield {
+      val correctNumberOfSigs = sign.cetSigs.outcomeSigs.size == client.outcomes.size
+
+      correctNumberOfSigs && sign.cetSigs.outcomeSigs.foldLeft(true) {
+        case (ret, (outcome, sig)) =>
+          ret && client.verifyCETSig(outcome, sig)
```

Comment:
ditto above for `ret`, and also you could do `sign.cetSigs.outcomeSigs.foldLeft(correctNumberOfSigs) { ...`

Reply:
This shouldn't be needed because scala has short circuiting implemented, so if the first part of the && is false, it won't evaluate the rest. Also with ||, if the first part is true then it won't evaluate the rest.